10_rest/willshen8/pkg/puppy/storer_test.go

```diff
@@ -0,0 +1,109 @@
+package puppy
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	out io.Writer = os.Stdout
+)
+
+type StorerSuite struct {
+	suite.Suite
+	storer Storer
+}
+
+func (s *StorerSuite) TestCreatePuppySuccessfully() {
+	m := assert.New(s.T())
+	puppy1 := Puppy{ID: 1, Breed: "Jack Russell Terrier", Color: "White and Brown", Value: "1500"}
+	createdID, _ := s.storer.CreatePuppy(&puppy1)
+	readPuppy1, _ := s.storer.ReadPuppy(createdID)
+	m.Equal(&puppy1, readPuppy1)
+}
+
+func (s *StorerSuite) TestCreatePuppyWithValueLessThanZero() {
+	InvalidPuppy := Puppy{ID: 22, Breed: "Pomeranian", Color: "White", Value: "-100"}
+	m := assert.New(s.T())
+	var buf bytes.Buffer
+	out = &buf
+	createdID, error := s.storer.CreatePuppy(&InvalidPuppy)
+	fmt.Fprintln(out, error)
+	m.Equal(ErrorCreateID, createdID)
+}
+
+func (s *StorerSuite) TestCreatePuppyWithWrongValue() {
+	m := assert.New(s.T())
+	WrongValuePuppy := Puppy{ID: 88, Breed: "Spaniel", Color: "Black", Value: "$$$"}
+	_, err := s.storer.CreatePuppy(&WrongValuePuppy)
+	m.Equal(&WrongValueError, err)
+}
+
+func (s *StorerSuite) TestCreateNextPuppyWithCorrectID() {
+	m := assert.New(s.T())
+	puppy2 := Puppy{ID: 1234, Breed: "Fox Terrier", Color: "Black", Value: "1300"}
+	puppy3 := Puppy{ID: 100, Breed: "German Shepperd", Color: "Brown", Value: "2000"}
+	createID1, creatError1 := s.storer.CreatePuppy(&puppy2)
+	if createID1 != uint32(1) && creatError1 != nil {
+		panic("Error creating puppy")
+	}
+	// create a second puppy should give an id of 2
+	createdID2, _ := s.storer.CreatePuppy(&puppy3)
+	m.Equal(uint32(2), createdID2)
+}
+
+func (s *StorerSuite) TestReadPuppySuccessfully() {
+	m := assert.New(s.T())
+	puppy4 := Puppy{ID: 120, Breed: "Golden Retriever", Color: "Golden", Value: "2500"}
+	returnedPuppyID4, _ := s.storer.CreatePuppy(&puppy4)
+	returnPuppy4, _ := s.storer.ReadPuppy(returnedPuppyID4)
+	m.Equal(&puppy4, returnPuppy4)
+}
+
+func (s *StorerSuite) TestReadNonExistentPuppy() {
+	m := assert.New(s.T())
+	returnPuppy, err := s.storer.ReadPuppy(100)
+	m.Nil(returnPuppy)
+	m.Equal(&ReadError, err)
+}
+
+func (s *StorerSuite) TestUpdatePuppySuccessfully() {
+	puppy5 := Puppy{ID: 200, Breed: "Chihuahua", Color: "White", Value: "500"}
+	puppy6 := Puppy{ID: 300, Breed: "Husky", Color: "White", Value: "3500"}
+	returnedPuppyID, _ := s.storer.CreatePuppy(&puppy5)
+	updateResult, _ := s.storer.UpdatePuppy(returnedPuppyID, &puppy6)
+	s.Equal(true, updateResult)
+}
+
+func (s *StorerSuite) TestUpdateNonExistentID() {
+	puppy1 := Puppy{ID: 1, Breed: "Jack Russell Terrier", Color: "White and Brown", Value: "1500"}
+	updateResult, error := s.storer.UpdatePuppy(888, &puppy1)
+	s.Equal(&UpdateError, error)
```

Comment:
interface is overwritten here. please `s/error/err/`

Reply:
Done